extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -211,7 +218,7 @@ function createDicomWebApi(dicomWebConfig, userAuthenticationService) {
     },
 
     store: {
-      dicom: async (dataset, request) => {
+      dicom: async (dataset, request, measurementData) => {
```

Comment:
I have some concerns about the current approach. It seems a bit unusual to have measurementData mentioned here. Ideally, the method that calls this should have already done its setup. My proposal is to create a dicomDict in the commandsModule of the SR and pass it here. So, it would look like this:

dicom: async(dataset, request, dicomDict)

Then, you can check if a dict exists and use it instead of creating a new one.